pkg/core/blockchain.go

```diff
@@ -677,36 +677,111 @@ func (bc *Blockchain) GetStateModule() blockchainer.StateRoot {
 // transactions with all appropriate side-effects and updates Blockchain state.
 // This is the only way to change Blockchain state.
 func (bc *Blockchain) storeBlock(block *block.Block, txpool *mempool.Pool) error {
-	cache := dao.NewCached(bc.dao)
-	writeBuf := io.NewBufBinWriter()
+	cache := bc.dao.GetWrapped()
 	appExecResults := make([]*state.AppExecResult, 0, 2+len(block.Transactions))
-	if err := cache.StoreAsBlock(block, writeBuf); err != nil {
-		return err
-	}
-	writeBuf.Reset()
 
-	if err := cache.StoreAsCurrentBlock(block, writeBuf); err != nil {
-		return err
-	}
-	writeBuf.Reset()
+	aerchan := make(chan *state.AppExecResult, len(block.Transactions)/8) // Tested 8 and 4, the difference is minimal, but free to test and tune.
+	aerdone := make(chan error)
+	blockdone := make(chan error)
+	go func() {
+		kvcache := cache.GetWrapped()
+		var writeBuf = io.NewBufBinWriter()
```

Comment:
Let's stick with `:=`?

Reply:
I don't like `:=` to be fair, so I'd rather make everything `var`.

Reply:
Fixed.